services/spar/src/Spar/Data.hs

```diff
@@ -212,6 +213,14 @@ insertSAMLUser (SAML.UserRef tenant subject) uid = retry x5 . write ins $ params
     ins :: PrepQuery W (SAML.Issuer, SAML.NameID, UserId) ()
     ins = "INSERT INTO user (issuer, sso_id, uid) VALUES (?, ?, ?)"
 
+-- | We only need to know if it's none or more, so this function returns the first one.
+getSAMLAnyUserByIssuer :: (HasCallStack, MonadClient m) => SAML.Issuer -> m (Maybe UserId)
+getSAMLAnyUserByIssuer issuer = runIdentity <$$>
```

Comment:
@fisx please change this C* query to use `LIMIT 1` otherwise you risk returning a potentially unbound nr. of users.

Reply:
https://github.com/wireapp/wire-server/pull/942

good catch, glad you've paid attention!  thanks :)